### PR TITLE
fix: post watcher not triggered (fix #12664)

### LINKED ISSUE #12664: Template Ref watcher is not triggered with `flush: post` option in Vue 2.7.
### Version
2.7.6

### Reproduction link
[codesandbox.io](https://codesandbox.io/embed/vue-2-7-flush-post-option-n54wb7?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
Open the CodeSandbox and watch the console output.
There seems to be a problem when Template Ref watcher components are nested.

### What is expected?
There should be a log entry for `App.vue`.

### What is actually happening?
There's no log entry because the watcher is not triggered.

---
This works in Vue 3.

https://codesandbox.io/embed/modest-bash-u84i9q?fontsize=14&hidenavigation=1&theme=dark

### PR DESCRIPTION
**Pull request by JuniorTour**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Fix #12664

### CODE REVIEW

File: src/core/observer/scheduler.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ function callActivatedHooks(queue) {
  */
 export function queueWatcher(watcher: Watcher) {
   const id = watcher.id
-  if (has[id] != null) {
+  if (id !== Infinity && has[id] != null) {
```
Comment (JuniorTour) [inline src/core/observer/scheduler.ts:159]: `Infinity` id comes from: https://github.com/vuejs/vue/blob/7d4a7722e3a8bc52a0a9b9660b75efa2ff479f39/src/v3/apiWatch.ts#L316

File: .gitignore [hunk 2]
```diff
@@ -12,6 +12,7 @@ packages/server-renderer/client-plugin.js
 packages/template-compiler/build.js
 packages/template-compiler/browser.js
 .vscode
+.idea
```
Comment (yyx990803) [inline .gitignore:15]:
```suggestion
```

Comment (yyx990803) [reply .gitignore]: ^ do not include unrelated changes in a PR.

**Comment by yyx990803** [comment]
Thanks for the PR. After looking at the original issue, I think a better fix is to avoid using `Infinity` as id. See 25ffdb62

**Pull request closed without merging**